recipes/xerces-c/all/conanfile.py

```diff
@@ -111,7 +114,7 @@ def _configure_cmake(self):
         self._cmake.definitions["xmlch-type"] = self.options.char_type
         self._cmake.definitions["mutex-manager"] = self.options.mutex_manager
         # avoid picking up system dependency
-        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_CURL"] = self.options.network_accessor != "curl"
+        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_CURL"] = True
```

Comment:
This seems a little odd how does the Conan provided curl get pulled into the project... I was not able to find it looking quickly in https://gitbox.apache.org/repos/asf?p=xerces-c.git;a=tree;f=cmake;h=2537654adeb129afd6e6c1b0d7d807c4df60649c;hb=HEAD... can you please point me in the right direction

Reply:
I thought conan would handle this automatically in the case of `self.requires("libcurl/7.78.0")`. Because of the comment above (*'avoid picking up system dependency'*) I thought that I would need to set `CMAKE_DISABLE_FIND_PACKAGE_CURL` to `True` in order for CMake to not use the system curl but rather the one provided by conan.  
But it seems I might be wrong in this regard...

Reply:
The way Conan works in general is its able to generate the usual build system files that a project requires.

Here with CMake theres a few option but most often `cmake_find_package_multi` is what's need to create the `FindPackage-config.cmake` file that's needed... 

if this project uses `find_package` then something might be a little off.

Reply:
It seems xerces-c uses `find_package` to pull in curl:

https://github.com/apache/xerces-c/blob/caa65157acf2a9e77a0bc5d9e7594a367198c440/cmake/XercesNetAccessorSelection.cmake#L34

I tried to build the conan package with the `network_accessor=curl` option now and found that `CMAKE_DISABLE_FIND_PACKAGE_CURL` needs to be `False` in this case or otherwise CMake won't find curl and the configure step will fail.  
Setting `self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_CURL"] = self.options.network_accessor != "curl"` then allowed me to build `libxerces-c.a` successfully but the build fails shortly after when building the included test applications due to linker errors:

<details>
<summary>Click to expand log</summary>

```
... snip ...

[ 80%] Linking CXX static library ../../lib/libxerces-c.a
[ 80%] Built target xerces-c
[ 81%] Building CXX object source_subfolder/tests/CMakeFiles/DOMTest.dir/src/DOM/DOMTest/DTest.cpp.o
[ 82%] Building CXX object source_subfolder/tests/CMakeFiles/XSValueTest.dir/src/XSValueTest/XSValueTest.cpp.o
[ 82%] Building CXX object source_subfolder/tests/CMakeFiles/DOMMemTest.dir/src/DOM/DOMMemTest/DOMMemTest.cpp.o
[ 82%] Building CXX object source_subfolder/tests/CMakeFiles/ThreadTest.dir/src/ThreadTest/ThreadTest.cpp.o
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1354:83:
warning: integer constant is so large that it is unsigned
 1354 |     XSValue::XSValue_Data act_v_ran64_v_2; act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                   ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1510:85:
warning: integer constant is so large that it is unsigned
 1510 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1658:85:
warning: integer constant is so large that it is unsigned
 1658 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1805:85:
warning: integer constant is so large that it is unsigned
 1805 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2427:94:
warning: integer constant is so large that it is unsigned
 2427 |     XSValue::XSValue_Data act_v_ran64_v_2;  act_v_ran64_v_2.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2579:94:
warning: integer constant is so large that it is unsigned
 2579 |     XSValue::XSValue_Data act_v_ran64_v_1;  act_v_ran64_v_1.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3194:94:
warning: integer constant is so large that it is unsigned
 3194 |     XSValue::XSValue_Data act_v_ran64_v_2;  act_v_ran64_v_2.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:29:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:29:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:44:6: warning: no previous declaration for 'void tassert(bool, const char*, int)' [-Wmissing-declarations]
   44 | void tassert(bool c, const char *file, int line)
      |      ^~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:119:6:
warning: no previous declaration for 'void DOMBasicTests()' [-Wmissing-declarations]
  119 | void DOMBasicTests()
      |      ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/internal/VecAttrListImpl.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/parsers/SAXParser.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:32:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/internal/VecAttrListImpl.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/parsers/SAXParser.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:32:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp: In function 'void DOMBasicTests()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:490:31
 warning: catching polymorphic type 'class xercesc_3_2::DOMException' by value [-Wcatch-value=]
  490 |          catch ( DOMException e)
      |                               ^
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.h:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.h:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:600:6:
warning: no previous declaration for 'void DOMNSTests()' [-Wmissing-declarations]
  600 | void DOMNSTests()
      |      ^~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.h: In member function 'virtual void myUserDataHandler::handle(xercesc_3_2::DOMUserDataHandler::DOMOperationType, const XMLCh*, void*, const xercesc_3_2::DOMNode*, xercesc_3_2::DOMNode*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.h:141:22: warning:
cast from type 'const XMLCh* const' {aka 'const short unsigned int* const'} to type 'XMLCh*' {aka 'short unsigned int*'} casts away qualifiers [-Wcast-qual]
  141 |         currentKey = (XMLCh*) key;
      |                      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.h:143:22: warning:
cast from type 'const xercesc_3_2::DOMNode*' to type 'xercesc_3_2::DOMNode*' casts away qualifiers [-Wcast-qual]
  143 |         currentSrc = (DOMNode*) src;
      |                      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:1398:6
 warning: no previous declaration for 'void DOMReleaseTests()' [-Wmissing-declarations]
 1398 | void DOMReleaseTests()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:1465:6
 warning: no previous declaration for 'int mymain()' [-Wmissing-declarations]
 1465 | int  mymain()
      |      ^~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: In static member function 'static void ThreadFuncs::startThread(ThreadFunc, void*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:100:39: warning: cast between incompatible function types from 'ThreadFunc' {aka 'void (*)(void*)'} to 'pthreadfunc' {aka 'void* (*)(void*)'} [-Wcast-function-type]
  100 |     x = pthread_create( &tId, &attr,  (pthreadfunc)func,  param);
      |                                       ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: In member function 'void ThreadParser::domCheckSum(const xercesc_3_2::DOMNode*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:707:12: warning: switch missing default case [-Wswitch-default]
  707 |     switch (node->getNodeType() )
      |            ^
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:823:6: warning: no previous declaration for 'void parseCommandLine(int, char**)' [-Wmissing-declarations]
  823 | void parseCommandLine(int argc, char **argv)
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:1009:6: warning: no previous declaration for 'void ReadFilesIntoMemory()' [-Wmissing-declarations]
 1009 | void ReadFilesIntoMemory()
      |      ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:1070:6: warning: no previous declaration for 'void threadMain(void*)' [-Wmissing-declarations]
 1070 | void threadMain (void *param)
      |      ^~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testDocument(xercesc_3_2::DOMDocument*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:2183:17: warning: variable 'newElement' set but not used [-Wunused-but-set-variable]
 2183 |     DOMElement* newElement;
      |                 ^~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testElement(xercesc_3_2::DOMDocument*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:3013:40: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 3013 |     DOMNodeList* text = ((DOMNode*  &) element2)->getChildNodes();
      |                                        ^~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testWholeText(xercesc_3_2::XercesDOMParser*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:4837:30: warning: cast from type 'const char*' to type 'XMLByte*' {aka 'unsigned char*'} casts away qualifiers [-Wcast-qual]
 4837 |         MemBufInputSource is((XMLByte*)sXml, strlen(sXml), "bufId");
      |                              ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testElementTraversal()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:5176:30: warning: cast from type 'const char*' to type 'XMLByte*' {aka 'unsigned char*'} casts away qualifiers [-Wcast-qual]
 5176 |         MemBufInputSource is((XMLByte*)sXml, strlen(sXml), "bufId");
      |                              ^~~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:27:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:706:7: warning: no previous declaration for 'void testNoActVal(const char*, xercesc_3_2::XSValue::DataType, xercesc_3_2::XSValue::Status)' [-Wmissing-declarations]
  706 | void  testNoActVal(const char*                  const  data
      |       ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:732:7: warning: no previous declaration for 'void testNoCanRep(const char*, xercesc_3_2::XSValue::DataType, xercesc_3_2::XSValue::Status)' [-Wmissing-declarations]
  732 | void  testNoCanRep(const char*                  const  data
      |       ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:760:6: warning: no previous declaration for 'void test_dt_decimal()' [-Wmissing-declarations]
  760 | void test_dt_decimal()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_decimal()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:790:27: warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
  790 |     XSValue::XSValue_Data act_v_ran32_v_2; act_v_ran32_v_2.fValue.f_double = (double)999.4294967295;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:942:6: warning: no previous declaration for 'void test_dt_float()' [-Wmissing-declarations]
  942 | void test_dt_float()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1135:6: warning: no previous declaration for 'void test_dt_double()' [-Wmissing-declarations]
 1135 | void test_dt_double()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1333:6: warning: no previous declaration for 'void test_dt_integer()' [-Wmissing-declarations]
 1333 | void test_dt_integer()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_integer()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1357:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 1357 |     XSValue::XSValue_Data act_v_ran32_v_2; act_v_ran32_v_2.fValue.f_long = (long)-2147483648;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1492:6: warning: no previous declaration for 'void test_dt_nonPositiveInteger()' [-Wmissing-declarations]
 1492 | void test_dt_nonPositiveInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1640:6: warning: no previous declaration for 'void test_dt_negativeInteger()' [-Wmissing-declarations]
 1640 | void test_dt_negativeInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1784:6: warning: no previous declaration for 'void test_dt_long()' [-Wmissing-declarations]
 1784 | void test_dt_long()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_long()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1808:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 1808 |     XSValue::XSValue_Data act_v_ran32_v_2;   act_v_ran32_v_2.fValue.f_long = (long)-2147483648;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1951:6: warning: no previous declaration for 'void test_dt_int()' [-Wmissing-declarations]
 1951 | void test_dt_int()
      |      ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2104:6: warning: no previous declaration for 'void test_dt_short()' [-Wmissing-declarations]
 2104 | void test_dt_short()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2258:6: warning: no previous declaration for 'void test_dt_byte()' [-Wmissing-declarations]
 2258 | void test_dt_byte()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2409:6: warning: no previous declaration for 'void test_dt_nonNegativeInteger()' [-Wmissing-declarations]
 2409 | void test_dt_nonNegativeInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2559:6: warning: no previous declaration for 'void test_dt_unsignedLong()' [-Wmissing-declarations]
 2559 | void test_dt_unsignedLong()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_unsignedLong()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2583:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 2583 |     XSValue::XSValue_Data act_v_ran32_v_2;  act_v_ran32_v_2.fValue.f_ulong = (unsigned long)0;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2727:6: warning: no previous declaration for 'void test_dt_unsignedInt()' [-Wmissing-declarations]
 2727 | void test_dt_unsignedInt()
      |      ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2877:6: warning: no previous declaration for 'void test_dt_unsignedShort()' [-Wmissing-declarations]
 2877 | void test_dt_unsignedShort()
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3027:6: warning: no previous declaration for 'void test_dt_unsignedByte()' [-Wmissing-declarations]
 3027 | void test_dt_unsignedByte()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3176:6: warning: no previous declaration for 'void test_dt_positiveInteger()' [-Wmissing-declarations]
 3176 | void test_dt_positiveInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3323:6: warning: no previous declaration for 'void test_dt_boolean()' [-Wmissing-declarations]
 3323 | void test_dt_boolean()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3441:6: warning: no previous declaration for 'void test_dt_hexBinary()' [-Wmissing-declarations]
 3441 | void test_dt_hexBinary()
      |      ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3559:6: warning: no previous declaration for 'void test_dt_base64Binary()' [-Wmissing-declarations]
 3559 | void test_dt_base64Binary()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3685:6: warning: no previous declaration for 'void test_dt_duration()' [-Wmissing-declarations]
 3685 | void test_dt_duration()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3820:6: warning: no previous declaration for 'void test_dt_date()' [-Wmissing-declarations]
 3820 | void test_dt_date()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4000:6: warning: no previous declaration for 'void test_dt_gYearMonth()' [-Wmissing-declarations]
 4000 | void test_dt_gYearMonth()
      |      ^~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4136:6: warning: no previous declaration for 'void test_dt_gYear()' [-Wmissing-declarations]
 4136 | void test_dt_gYear()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4273:6: warning: no previous declaration for 'void test_dt_gMonthDay()' [-Wmissing-declarations]
 4273 | void test_dt_gMonthDay()
      |      ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4409:6: warning: no previous declaration for 'void test_dt_gDay()' [-Wmissing-declarations]
 4409 | void test_dt_gDay()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4546:6: warning: no previous declaration for 'void test_dt_gMonth()' [-Wmissing-declarations]
 4546 | void test_dt_gMonth()
      |      ^~~~~~~~~~~~~~
[ 83%] Linking CXX executable ../../bin/ThreadTest
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4683:6: warning: no previous declaration for 'void test_dt_dateTime()' [-Wmissing-declarations]
 4683 | void test_dt_dateTime()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4849:6: warning: no previous declaration for 'void test_dt_time()' [-Wmissing-declarations]
 4849 | void test_dt_time()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5006:6: warning: no previous declaration for 'void test_dt_string()' [-Wmissing-declarations]
 5006 | void test_dt_string()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5092:6: warning: no previous declaration for 'void test_dt_anyURI()' [-Wmissing-declarations]
 5092 | void test_dt_anyURI()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5208:6: warning: no previous declaration for 'void test_dt_QName()' [-Wmissing-declarations]
 5208 | void test_dt_QName()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5324:6: warning: no previous declaration for 'void test_dt_NOTATION()' [-Wmissing-declarations]
 5324 | void test_dt_NOTATION()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5415:6: warning: no previous declaration for 'void test_dt_normalizedString()' [-Wmissing-declarations]
 5415 | void test_dt_normalizedString()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5526:6: warning: no previous declaration for 'void test_dt_token()' [-Wmissing-declarations]
 5526 | void test_dt_token()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_token()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5536:27:
warning: variable 'act_v_ran_v_2' set but not used [-Wunused-but-set-variable]
 5536 |     XSValue::XSValue_Data act_v_ran_v_2;  act_v_ran_v_2.fValue.f_strVal = 0;
      |                           ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5537:27:
warning: variable 'act_v_ran_v_3' set but not used [-Wunused-but-set-variable]
 5537 |     XSValue::XSValue_Data act_v_ran_v_3;  act_v_ran_v_3.fValue.f_strVal = 0;
      |                           ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5643:6: warning: no previous declaration for 'void test_dt_language()' [-Wmissing-declarations]
 5643 | void test_dt_language()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5759:6: warning: no previous declaration for 'void test_dt_NMTOKEN()' [-Wmissing-declarations]
 5759 | void test_dt_NMTOKEN()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5875:6: warning: no previous declaration for 'void test_dt_NMTOKENS()' [-Wmissing-declarations]
 5875 | void test_dt_NMTOKENS()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5992:6: warning: no previous declaration for 'void test_dt_Name()' [-Wmissing-declarations]
 5992 | void test_dt_Name()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6109:6: warning: no previous declaration for 'void test_dt_NCName_ID_IDREF_ENTITY(xercesc_3_2::XSValue::DataType)' [-Wmissing-declarations]
 6109 | void test_dt_NCName_ID_IDREF_ENTITY(XSValue::DataType dt)
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6224:6: warning: no previous declaration for 'void test_dt_IDREFS_ENTITIES(xercesc_3_2::XSValue::DataType)' [-Wmissing-declarations]
 6224 | void test_dt_IDREFS_ENTITIES(XSValue::DataType dt)
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6339:6: warning: no previous declaration for 'void test_DataType()' [-Wmissing-declarations]
 6339 | void test_DataType()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6390:6: warning: no previous declaration for 'void testErrorStatus()' [-Wmissing-declarations]
 6390 | void testErrorStatus()
      |      ^~~~~~~~~~~~~~~
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `deflate_init_writer':
content_encoding.c:(.text+0x1f4): undefined reference to `inflateInit_'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `deflate_close_writer':
content_encoding.c:(.text+0x2b5): undefined reference to `inflateEnd'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `gzip_close_writer':
content_encoding.c:(.text+0x345): undefined reference to `inflateEnd'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `gzip_init_writer':
content_encoding.c:(.text+0x3ca): undefined reference to `zlibVersion'
/usr/bin/ld: content_encoding.c:(.text+0x3f6): undefined reference to `inflateInit2_'
/usr/bin/ld: content_encoding.c:(.text+0x419): undefined reference to `inflateInit2_'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `inflate_stream':
content_encoding.c:(.text+0x4ed): undefined reference to `inflate'
/usr/bin/ld: content_encoding.c:(.text+0x66f): undefined reference to `inflateEnd'
/usr/bin/ld: content_encoding.c:(.text+0x6d4): undefined reference to `inflateEnd'
/usr/bin/ld: content_encoding.c:(.text+0x704): undefined reference to `inflateEnd'
/usr/bin/ld: content_encoding.c:(.text+0x71d): undefined reference to `inflateInit2_'
/usr/bin/ld: content_encoding.c:(.text+0x7b0): undefined reference to `inflateEnd'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `deflate_unencode_write':
content_encoding.c:(.text+0x885): undefined reference to `inflateEnd'
/usr/bin/ld: content_encoding.c:(.text+0x8ba): undefined reference to `inflateEnd'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o): in function `gzip_unencode_write':
content_encoding.c:(.text+0x9f0): undefined reference to `inflateEnd'
/usr/bin/ld: content_encoding.c:(.text+0xaba): undefined reference to `inflateEnd'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-content_encoding.o):content_encoding.c:(.text+0xb2c): more undefined references to `inflateEnd' follow
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o): in function `Curl_md5it':
md5.c:(.text+0x2b): undefined reference to `MD5_Init'
/usr/bin/ld: md5.c:(.text+0x40): undefined reference to `MD5_Update'
/usr/bin/ld: md5.c:(.text+0x4b): undefined reference to `MD5_Final'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x0): undefined reference to `MD5_Init'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x8): undefined reference to `MD5_Update'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x10): undefined reference to `MD5_Final'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x20): undefined reference to `MD5_Init'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x28): undefined reference to `MD5_Update'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-md5.o):(.data.rel.ro+0x30): undefined reference to `MD5_Final'
/usr/bin/ld: /home/florian/.conan/data/libcurl/7.78.0/_/_/package/ac2c4ce2ca2a9f41f0bdc349d469de280c20290f/lib/libcurl.a(libcurl_la-sha256.o): in function `Curl_sha256it':
sha256.c:(.text+0x2e): undefined reference to `SHA256_Init'

... snip ...

collect2: error: ld returned 1 exit status
make[2]: *** [source_subfolder/tests/CMakeFiles/XSValueTest.dir/build.make:100: bin/XSValueTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:356: source_subfolder/tests/CMakeFiles/XSValueTest.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
xerces-c/3.2.3:
xerces-c/3.2.3: ERROR: Package '7db810552686fb4c1a673142b9169bcef48cdd22' build failed
xerces-c/3.2.3: WARN: Build folder /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22
ERROR: xerces-c/3.2.3: Error in build() method, line 124
        cmake.build()
        ConanException: Error 2 while executing cmake --build '/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/7db810552686fb4c1a673142b9169bcef48cdd22/build_subfolder' '--' '-j4'
```
</details>

I also tried the same thing with the `transcoder=icu` option to see if the linking error problem might be specific to curl but the behaviour was similar:

<details>
<summary> Click to expand log </summary>

```
... snip ...

[ 80%] Linking CXX static library ../../lib/libxerces-c.a
[ 80%] Built target xerces-c
[ 81%] Building CXX object source_subfolder/tests/CMakeFiles/ThreadTest.dir/src/ThreadTest/ThreadTest.cpp.o
[ 81%] Building CXX object source_subfolder/tests/CMakeFiles/XSValueTest.dir/src/XSValueTest/XSValueTest.cpp.o
[ 81%] Building CXX object source_subfolder/tests/CMakeFiles/DOMMemTest.dir/src/DOM/DOMMemTest/DOMMemTest.cpp.o
[ 81%] Building CXX object source_subfolder/tests/CMakeFiles/DOMTest.dir/src/DOM/DOMTest/DTest.cpp.o
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1354:83:
warning: integer constant is so large that it is unsigned
 1354 |     XSValue::XSValue_Data act_v_ran64_v_2; act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                   ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1510:85:
warning: integer constant is so large that it is unsigned
 1510 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1658:85:
warning: integer constant is so large that it is unsigned
 1658 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1805:85:
warning: integer constant is so large that it is unsigned
 1805 |     XSValue::XSValue_Data act_v_ran64_v_2;   act_v_ran64_v_2.fValue.f_long = (long)-9223372036854775808;
      |                                                                                     ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2427:94:
warning: integer constant is so large that it is unsigned
 2427 |     XSValue::XSValue_Data act_v_ran64_v_2;  act_v_ran64_v_2.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2579:94:
warning: integer constant is so large that it is unsigned
 2579 |     XSValue::XSValue_Data act_v_ran64_v_1;  act_v_ran64_v_1.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3194:94:
warning: integer constant is so large that it is unsigned
 3194 |     XSValue::XSValue_Data act_v_ran64_v_2;  act_v_ran64_v_2.fValue.f_ulong = (unsigned long)+18446744073709551615;
      |                                                                                              ^~~~~~~~~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/internal/VecAttrListImpl.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/parsers/SAXParser.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:32:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/internal/VecAttrListImpl.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/parsers/SAXParser.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:32:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:29:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:29:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.h:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSObject.hpp:26,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSSimpleTypeDefinition.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/datatype/DatatypeValidator.hpp:32,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLAttr.hpp:28,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/XMLElementDecl.hpp:25,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/common/Grammar.hpp:27,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOMLSParser.hpp:29,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/dom/DOM.hpp:59,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.h:30,
                 from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/framework/psvi/XSConstants.hpp:56:24: warning: 'class xercesc_3_2::XSConstants' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   56 | class XMLPARSER_EXPORT XSConstants
      |                        ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:44:6: warning: no previous declaration for 'void tassert(bool, const char*, int)' [-Wmissing-declarations]
   44 | void tassert(bool c, const char *file, int line)
      |      ^~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:119:6:
warning: no previous declaration for 'void DOMBasicTests()' [-Wmissing-declarations]
  119 | void DOMBasicTests()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp: In function 'void DOMBasicTests()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:490:31
 warning: catching polymorphic type 'class xercesc_3_2::DOMException' by value [-Wcatch-value=]
  490 |          catch ( DOMException e)
      |                               ^
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:600:6:
warning: no previous declaration for 'void DOMNSTests()' [-Wmissing-declarations]
  600 | void DOMNSTests()
      |      ^~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:31:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.h: In member function 'virtual void myUserDataHandler::handle(xercesc_3_2::DOMUserDataHandler::DOMOperationType, const XMLCh*, void*, const xercesc_3_2::DOMNode*, xercesc_3_2::DOMNode*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.h:141:22: warning:
cast from type 'const XMLCh* const' {aka 'const short unsigned int* const'} to type 'XMLCh*' {aka 'short unsigned int*'} casts away qualifiers [-Wcast-qual]
  141 |         currentKey = (XMLCh*) key;
      |                      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.h:143:22: warning:
cast from type 'const xercesc_3_2::DOMNode*' to type 'xercesc_3_2::DOMNode*' casts away qualifiers [-Wcast-qual]
  143 |         currentSrc = (DOMNode*) src;
      |                      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:1398:6
 warning: no previous declaration for 'void DOMReleaseTests()' [-Wmissing-declarations]
 1398 | void DOMReleaseTests()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMMemTest/DOMMemTest.cpp:1465:6
 warning: no previous declaration for 'int mymain()' [-Wmissing-declarations]
 1465 | int  mymain()
      |      ^~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: In static member function 'static void ThreadFuncs::startThread(ThreadFunc, void*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:100:39: warning: cast between incompatible function types from 'ThreadFunc' {aka 'void (*)(void*)'} to 'pthreadfunc' {aka 'void* (*)(void*)'} [-Wcast-function-type]
  100 |     x = pthread_create( &tId, &attr,  (pthreadfunc)func,  param);
      |                                       ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: In member function 'void ThreadParser::domCheckSum(const xercesc_3_2::DOMNode*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:707:12: warning: switch missing default case [-Wswitch-default]
  707 |     switch (node->getNodeType() )
      |            ^
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:823:6: warning: no previous declaration for 'void parseCommandLine(int, char**)' [-Wmissing-declarations]
  823 | void parseCommandLine(int argc, char **argv)
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:1009:6: warning: no previous declaration for 'void ReadFilesIntoMemory()' [-Wmissing-declarations]
 1009 | void ReadFilesIntoMemory()
      |      ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/ThreadTest/ThreadTest.cpp:1070:6: warning: no previous declaration for 'void threadMain(void*)' [-Wmissing-declarations]
 1070 | void threadMain (void *param)
      |      ^~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testDocument(xercesc_3_2::DOMDocument*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:2183:17: warning: variable 'newElement' set but not used [-Wunused-but-set-variable]
 2183 |     DOMElement* newElement;
      |                 ^~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testElement(xercesc_3_2::DOMDocument*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:3013:40: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
 3013 |     DOMNodeList* text = ((DOMNode*  &) element2)->getChildNodes();
      |                                        ^~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testWholeText(xercesc_3_2::XercesDOMParser*)':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:4837:30: warning: cast from type 'const char*' to type 'XMLByte*' {aka 'unsigned char*'} casts away qualifiers [-Wcast-qual]
 4837 |         MemBufInputSource is((XMLByte*)sXml, strlen(sXml), "bufId");
      |                              ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp: In member function 'bool DOMTest::testElementTraversal()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/DOM/DOMTest/DTest.cpp:5176:30: warning: cast from type 'const char*' to type 'XMLByte*' {aka 'unsigned char*'} casts away qualifiers [-Wcast-qual]
 5176 |         MemBufInputSource is((XMLByte*)sXml, strlen(sXml), "bufId");
      |                              ^~~~~~~~~~~~~~
In file included from /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:27:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/src/xercesc/validators/schema/SchemaSymbols.hpp:33:25: warning: 'class xercesc_3_2::SchemaSymbols' only defines private constructors and has no friends [-Wctor-dtor-privacy]
   33 | class VALIDATORS_EXPORT SchemaSymbols
      |                         ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:706:7: warning: no previous declaration for 'void testNoActVal(const char*, xercesc_3_2::XSValue::DataType, xercesc_3_2::XSValue::Status)' [-Wmissing-declarations]
  706 | void  testNoActVal(const char*                  const  data
      |       ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:732:7: warning: no previous declaration for 'void testNoCanRep(const char*, xercesc_3_2::XSValue::DataType, xercesc_3_2::XSValue::Status)' [-Wmissing-declarations]
  732 | void  testNoCanRep(const char*                  const  data
      |       ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:760:6: warning: no previous declaration for 'void test_dt_decimal()' [-Wmissing-declarations]
  760 | void test_dt_decimal()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_decimal()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:790:27: warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
  790 |     XSValue::XSValue_Data act_v_ran32_v_2; act_v_ran32_v_2.fValue.f_double = (double)999.4294967295;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:942:6: warning: no previous declaration for 'void test_dt_float()' [-Wmissing-declarations]
  942 | void test_dt_float()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1135:6: warning: no previous declaration for 'void test_dt_double()' [-Wmissing-declarations]
 1135 | void test_dt_double()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1333:6: warning: no previous declaration for 'void test_dt_integer()' [-Wmissing-declarations]
 1333 | void test_dt_integer()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_integer()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1357:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 1357 |     XSValue::XSValue_Data act_v_ran32_v_2; act_v_ran32_v_2.fValue.f_long = (long)-2147483648;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1492:6: warning: no previous declaration for 'void test_dt_nonPositiveInteger()' [-Wmissing-declarations]
 1492 | void test_dt_nonPositiveInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1640:6: warning: no previous declaration for 'void test_dt_negativeInteger()' [-Wmissing-declarations]
 1640 | void test_dt_negativeInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1784:6: warning: no previous declaration for 'void test_dt_long()' [-Wmissing-declarations]
 1784 | void test_dt_long()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_long()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1808:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 1808 |     XSValue::XSValue_Data act_v_ran32_v_2;   act_v_ran32_v_2.fValue.f_long = (long)-2147483648;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:1951:6: warning: no previous declaration for 'void test_dt_int()' [-Wmissing-declarations]
 1951 | void test_dt_int()
      |      ^~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2104:6: warning: no previous declaration for 'void test_dt_short()' [-Wmissing-declarations]
 2104 | void test_dt_short()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2258:6: warning: no previous declaration for 'void test_dt_byte()' [-Wmissing-declarations]
 2258 | void test_dt_byte()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2409:6: warning: no previous declaration for 'void test_dt_nonNegativeInteger()' [-Wmissing-declarations]
 2409 | void test_dt_nonNegativeInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2559:6: warning: no previous declaration for 'void test_dt_unsignedLong()' [-Wmissing-declarations]
 2559 | void test_dt_unsignedLong()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_unsignedLong()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2583:27:
warning: variable 'act_v_ran32_v_2' set but not used [-Wunused-but-set-variable]
 2583 |     XSValue::XSValue_Data act_v_ran32_v_2;  act_v_ran32_v_2.fValue.f_ulong = (unsigned long)0;
      |                           ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2727:6: warning: no previous declaration for 'void test_dt_unsignedInt()' [-Wmissing-declarations]
 2727 | void test_dt_unsignedInt()
      |      ^~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:2877:6: warning: no previous declaration for 'void test_dt_unsignedShort()' [-Wmissing-declarations]
 2877 | void test_dt_unsignedShort()
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3027:6: warning: no previous declaration for 'void test_dt_unsignedByte()' [-Wmissing-declarations]
 3027 | void test_dt_unsignedByte()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3176:6: warning: no previous declaration for 'void test_dt_positiveInteger()' [-Wmissing-declarations]
 3176 | void test_dt_positiveInteger()
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3323:6: warning: no previous declaration for 'void test_dt_boolean()' [-Wmissing-declarations]
 3323 | void test_dt_boolean()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3441:6: warning: no previous declaration for 'void test_dt_hexBinary()' [-Wmissing-declarations]
 3441 | void test_dt_hexBinary()
      |      ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3559:6: warning: no previous declaration for 'void test_dt_base64Binary()' [-Wmissing-declarations]
 3559 | void test_dt_base64Binary()
      |      ^~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3685:6: warning: no previous declaration for 'void test_dt_duration()' [-Wmissing-declarations]
 3685 | void test_dt_duration()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:3820:6: warning: no previous declaration for 'void test_dt_date()' [-Wmissing-declarations]
 3820 | void test_dt_date()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4000:6: warning: no previous declaration for 'void test_dt_gYearMonth()' [-Wmissing-declarations]
 4000 | void test_dt_gYearMonth()
      |      ^~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4136:6: warning: no previous declaration for 'void test_dt_gYear()' [-Wmissing-declarations]
 4136 | void test_dt_gYear()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4273:6: warning: no previous declaration for 'void test_dt_gMonthDay()' [-Wmissing-declarations]
 4273 | void test_dt_gMonthDay()
      |      ^~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4409:6: warning: no previous declaration for 'void test_dt_gDay()' [-Wmissing-declarations]
 4409 | void test_dt_gDay()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4546:6: warning: no previous declaration for 'void test_dt_gMonth()' [-Wmissing-declarations]
 4546 | void test_dt_gMonth()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4683:6: warning: no previous declaration for 'void test_dt_dateTime()' [-Wmissing-declarations]
 4683 | void test_dt_dateTime()
      |      ^~~~~~~~~~~~~~~~
[ 82%] Linking CXX executable ../../bin/ThreadTest
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:4849:6: warning: no previous declaration for 'void test_dt_time()' [-Wmissing-declarations]
 4849 | void test_dt_time()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5006:6: warning: no previous declaration for 'void test_dt_string()' [-Wmissing-declarations]
 5006 | void test_dt_string()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5092:6: warning: no previous declaration for 'void test_dt_anyURI()' [-Wmissing-declarations]
 5092 | void test_dt_anyURI()
      |      ^~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5208:6: warning: no previous declaration for 'void test_dt_QName()' [-Wmissing-declarations]
 5208 | void test_dt_QName()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5324:6: warning: no previous declaration for 'void test_dt_NOTATION()' [-Wmissing-declarations]
 5324 | void test_dt_NOTATION()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5415:6: warning: no previous declaration for 'void test_dt_normalizedString()' [-Wmissing-declarations]
 5415 | void test_dt_normalizedString()
      |      ^~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5526:6: warning: no previous declaration for 'void test_dt_token()' [-Wmissing-declarations]
 5526 | void test_dt_token()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: In function 'void test_dt_token()':
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5536:27:
warning: variable 'act_v_ran_v_2' set but not used [-Wunused-but-set-variable]
 5536 |     XSValue::XSValue_Data act_v_ran_v_2;  act_v_ran_v_2.fValue.f_strVal = 0;
      |                           ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5537:27:
warning: variable 'act_v_ran_v_3' set but not used [-Wunused-but-set-variable]
 5537 |     XSValue::XSValue_Data act_v_ran_v_3;  act_v_ran_v_3.fValue.f_strVal = 0;
      |                           ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp: At global scope:
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5643:6: warning: no previous declaration for 'void test_dt_language()' [-Wmissing-declarations]
 5643 | void test_dt_language()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5759:6: warning: no previous declaration for 'void test_dt_NMTOKEN()' [-Wmissing-declarations]
 5759 | void test_dt_NMTOKEN()
      |      ^~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5875:6: warning: no previous declaration for 'void test_dt_NMTOKENS()' [-Wmissing-declarations]
 5875 | void test_dt_NMTOKENS()
      |      ^~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:5992:6: warning: no previous declaration for 'void test_dt_Name()' [-Wmissing-declarations]
 5992 | void test_dt_Name()
      |      ^~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6109:6: warning: no previous declaration for 'void test_dt_NCName_ID_IDREF_ENTITY(xercesc_3_2::XSValue::DataType)' [-Wmissing-declarations]
 6109 | void test_dt_NCName_ID_IDREF_ENTITY(XSValue::DataType dt)
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6224:6: warning: no previous declaration for 'void test_dt_IDREFS_ENTITIES(xercesc_3_2::XSValue::DataType)' [-Wmissing-declarations]
 6224 | void test_dt_IDREFS_ENTITIES(XSValue::DataType dt)
      |      ^~~~~~~~~~~~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6339:6: warning: no previous declaration for 'void test_DataType()' [-Wmissing-declarations]
 6339 | void test_DataType()
      |      ^~~~~~~~~~~~~
/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/source_subfolder/tests/src/XSValueTest/XSValueTest.cpp:6390:6: warning: no previous declaration for 'void testErrorStatus()' [-Wmissing-declarations]
 6390 | void testErrorStatus()
      |      ^~~~~~~~~~~~~~~
/usr/bin/ld: /home/florian/.conan/data/icu/69.1/_/_/package/53570d5caa8376dbdead2f216b76930a34958133/lib/libicuuc.a(putil.ao): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [source_subfolder/tests/CMakeFiles/ThreadTest.dir/build.make:101: bin/ThreadTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:408: source_subfolder/tests/CMakeFiles/ThreadTest.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 82%] Linking CXX executable ../../bin/DOMMemTest
/usr/bin/ld: /home/florian/.conan/data/icu/69.1/_/_/package/53570d5caa8376dbdead2f216b76930a34958133/lib/libicuuc.a(putil.ao): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [source_subfolder/tests/CMakeFiles/DOMMemTest.dir/build.make:101: bin/DOMMemTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:434: source_subfolder/tests/CMakeFiles/DOMMemTest.dir/all] Error 2
[ 82%] Linking CXX executable ../../bin/DOMTest
/usr/bin/ld: /home/florian/.conan/data/icu/69.1/_/_/package/53570d5caa8376dbdead2f216b76930a34958133/lib/libicuuc.a(putil.ao): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [source_subfolder/tests/CMakeFiles/DOMTest.dir/build.make:101: bin/DOMTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:382: source_subfolder/tests/CMakeFiles/DOMTest.dir/all] Error 2
[ 83%] Linking CXX executable ../../bin/XSValueTest
/usr/bin/ld: /home/florian/.conan/data/icu/69.1/_/_/package/53570d5caa8376dbdead2f216b76930a34958133/lib/libicuuc.a(putil.ao): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [source_subfolder/tests/CMakeFiles/XSValueTest.dir/build.make:101: bin/XSValueTest] Error 1
make[1]: *** [CMakeFiles/Makefile2:356: source_subfolder/tests/CMakeFiles/XSValueTest.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
xerces-c/3.2.3:
xerces-c/3.2.3: ERROR: Package 'c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd' build failed
xerces-c/3.2.3: WARN: Build folder /home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd
ERROR: xerces-c/3.2.3: Error in build() method, line 124
        cmake.build()
        ConanException: Error 2 while executing cmake --build '/home/florian/.conan/data/xerces-c/3.2.3/_/_/build/c264a178dfebc4dc25a69bcec7d9eb97ff11b1fd/build_subfolder' '--' '-j4'
```

</details>

Frankly, I'm a bit lost right now and don't really know where these errors come from or how I can resolve them. (Building xerces-c manually with CMake using the system ICU/curl suceeds.) Maybe you have an idea?

Reply:
Building everything as shared libraries seems to work though (no build errors, test_package succeeds and I can run the test applications)

Reply:
It's most likely picking up your system libraries.

On my phone but the logs look like test code... which we do not compile in CCI so I am not sure what the issue might me.

I have a long weekend coming up so I will have time and I'll try to look deeper if you dont mind waiting.

I see CURL is only valid on linux, do we have a check for that in validate?

Reply:
So in short... the issue is that if you want `find_package` to work within conan you need to add the right generators... previous it was using the deprecated https://github.com/conan-io/conan-center-index/blob/4fcae3b4dfe4231366bbe082aac11f5e85944334/recipes/xerces-c/all/conanfile.py#L16

I've opened a PR against ur repo with my fixes https://github.com/FMeinicke/conan-center-index/pull/1

```
-- Xerces-C++ configuration summary
-- --------------------------------
--
--   Version:                   3.2.3
--   Library interface version: 3.2
--
--   Installation directory:    /home/cmcarthur/.conan/data/xerces-c/3.2.3/_/_/package/8ce0ba4e4863d6dd8374903f568b69d5d256f3ff
--   C compiler:                /usr/bin/cc
--   C++ compiler:              /usr/bin/c++
--
--   Build shared libraries:    OFF
--   Path delimiters:           "/"
--   File Manager:              POSIX
--   Mutex Manager:             posix
--   Transcoder:                icu
--   NetAccessor:               curl
--   Message Loader:            inmemory
--   XMLCh type:                uint16_t
```

And My build passes locally...

Reply:
Thank you again for taking the time to look into this!

I just tested FMeinicke/conan-center-index#1 locally and it works for me as well.